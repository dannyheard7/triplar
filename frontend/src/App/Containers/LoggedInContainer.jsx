import React from "react";
import {connect} from "react-redux";
import {withRouter} from "react-router-dom";
import axios from "axios";

export class LoggedInContainer extends React.Component {
    constructor(props) {
        super(props);

        if (this.props.isAuthenticated) {
            this.setAuthHeader(this.props.token);
        }
    }

    componentWillReceiveProps(nextProps){
        if (nextProps.isAuthenticated) {
            this.setAuthHeader(nextProps.token);
        }
    }

    setAuthHeader(token) {
        axios.defaults.headers.common = {
            'Authorization': 'JWT ' + token
        };
    }

    render() {
        return (
            <div>
                {this.props.children}
            </div>
        )
    }
}


function mapStateToProps(state, props) {
    return {
        isAuthenticated: state.auth.isAuthenticated,
        token: state.auth.token
    }
}

export default withRouter(connect(mapStateToProps)(LoggedInContainer))