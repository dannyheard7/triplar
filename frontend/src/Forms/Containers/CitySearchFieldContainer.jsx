import React from "react";
import api from "utils/api.js";
import AutosuggestFieldGroup from "Forms/Components/AutosuggestFieldGroup";

export default class CitySearchFieldContainer extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            value: this.props.value,
            suggestions: []
        };

        this.onChange = this.onChange.bind(this);
    }

    getSuggestions = (event) => {
        let value = event.value;

        api.searchCities(value).then(response => {
            if(response.status === 200) {
                this.setState({suggestions: response.data})
            }
        });
    };

    onSuggestionsClearRequested = () => {
        this.setState({suggestions: []});
    };

    shouldRenderSuggestions(value) {
      return value.trim().length > 2;
    }

    getSuggestionValue = suggestion => suggestion.name + ", " + suggestion.country;
    renderSuggestion(suggestion) {
         return (
             <div>
                 {suggestion.name + ", " + suggestion.country}
             </div>
         );
    }

    onChange(event, { newValue }) {
         this.setState({value: newValue});
     };


    render() {
        return (
            <AutosuggestFieldGroup name={this.props.name} value={this.state.value} suggestions={this.state.suggestions}
                                   renderSuggestion={this.renderSuggestion} getSuggestionValue={this.getSuggestionValue}
                                   getSuggestions={this.getSuggestions} onSuggestionsClearRequested={this.onSuggestionsClearRequested}
                                shouldRenderSuggestions={this.shouldRenderSuggestions} label="Location" errors={this.props.errors}
                                onChange={this.onChange}/>
        );
    }
}