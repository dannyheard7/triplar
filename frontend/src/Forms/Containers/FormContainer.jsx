import React from "react";
import {inputParsers} from "Forms/utils/inputParsers.js";

export default class FormContainer extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            errors: []
        };

        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleSubmit(event) {
        event.preventDefault();
        const data = this.parseFormData(event);

        this.props.apiFunction(data).then(response => {
            if (response.status === 400) {
                this.setState({errors: response.data})
            } else {
                this.setState({errors: []});
                this.props.onSuccess(response.data);
            }
        });
    }

    parseFormData(event) {
        const form = event.target;
        const data = new FormData(event.target);

        for (let name of data.keys()) {
            const input = form.elements[name];
            const parserName = input.dataset.parse;

            if (parserName) {
                const parser = inputParsers[parserName];
                const parsedValue = parser(data.get(name));
                data.set(name, parsedValue);
            }
        }

        return data;
    }

    render() {
        const props = {
            onSubmit: this.handleSubmit,
            errors: this.state.errors
        };

        return React.cloneElement(this.props.children, {...props});
    }

}