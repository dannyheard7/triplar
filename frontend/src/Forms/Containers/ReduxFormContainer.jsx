import React from "react";
import {inputParsers} from "../utils/inputParsers.js";
import {connect} from "react-redux";

class ReduxFormContainer extends React.Component {
    constructor(props) {
        super(props);

        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleSubmit(event) {
        event.preventDefault();
        const data = this.parseFormData(event);

        this.props.dispatch(this.props.action(data));
    }

    parseFormData(event) {
        const form = event.target;
        const data = new FormData(event.target);

        for (let name of data.keys()) {
            const input = form.elements[name];
            const parserName = input.dataset.parse;

            if (parserName) {
                const parser = inputParsers[parserName];
                const parsedValue = parser(data.get(name));
                data.set(name, parsedValue);
            }
        }

        let variables = Object();
        for(let pair of data.entries()) {
           variables[pair[0]] = pair[1];
        }

        return variables;
    }

    render() {
        const props = {
            onSubmit: this.handleSubmit,
            fieldErrors: this.props.fieldErrors,
            nonFieldErrors: this.props.nonFieldErrors

        };

        return React.cloneElement(this.props.children, {...props});
    }
}

ReduxFormContainer.defaultProps = {
    fieldErrors: [],
    nonFieldErrors: []
}

export default connect()(ReduxFormContainer)