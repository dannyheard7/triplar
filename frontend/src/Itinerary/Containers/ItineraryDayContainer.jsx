import React from "react";
import ItineraryDay from "Itinerary/Components/ItineraryDay";
import DroppablePlaceListContainer from "Places/Containers/DroppablePlaceListContainer";
import { connect } from "react-redux";
import {addItemToItineraryDay} from "../utils/actions";
import MarkerMap from "../../Maps/Components/MarkerMap";

export class ItineraryDayContainer extends React.Component {
    constructor(props) {
        super(props);

        this.canDrop = this.canDrop.bind(this);
        this.onDrop = this.onDrop.bind(this);
        this.onDragRemove = this.onDragRemove.bind(this);
    }

    canDrop(item) {
        return (this.props.places.indexOf(item) === -1)
    }

    onDrop(item) {
        this.props.dispatch(addItemToItineraryDay(this.props.itinerary.id, item.id, this.props.day, this.props.places.length));
    }

    onDragRemove(item) {
        console.log("here");
    }

    render () {
        const itinerary = this.props.itinerary;
        const position = [itinerary.city.location.lat, itinerary.city.location.lng];
        let placePositions = this.props.places.filter(x => x.coordinates).map(x => ({position: [x.coordinates.latitude, x.coordinates.longitude],
            popupText: x.name}));

        return (
            <div>
                <MarkerMap center={position} zoom={13} markers={placePositions}/>
                <ItineraryDay day={this.props.day}/>
                <DroppablePlaceListContainer places={this.props.places} path={this.props.path} canDrop={this.canDrop}
                                             onDrop={this.onDrop} onDragRemove={this.onDragRemove}/>
            </div>
        );
    }
}

function mapStateToProps(state, ownProps) {
    let placeIds = state.itineraries.itineraryItems.filter(x => x.itineraryId === ownProps.itinerary.id &&
        x.date === ownProps.day).map(x => x['place']);

    return {
        places: state.places.places.filter(p => placeIds.includes(p.id))
    }
}

export default connect(mapStateToProps)(ItineraryDayContainer)
