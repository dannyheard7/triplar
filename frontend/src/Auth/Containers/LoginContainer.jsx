import React from "react";
import api from "utils/api.js";
import {connect} from "react-redux";
import {withRouter} from "react-router-dom";

import {login} from "Auth/actions";
import LoginForm from "Auth/Components/LoginForm.jsx";
import FormContainer from "Forms/Containers/FormContainer";

export class LoginContainer extends React.Component {
    constructor(props) {
        super(props);

        this.onSuccess = this.onSuccess.bind(this);
    }

    onSuccess(data) {
        this.props.dispatch(login(data.user, data.token));
        this.props.history.push('/trips')
    }

    render() {
        return (
            <FormContainer onSuccess={this.onSuccess} apiFunction={api.getLoginToken}>
                <LoginForm />
            </FormContainer>
        );
    }
}

export default withRouter(connect()(LoginContainer))