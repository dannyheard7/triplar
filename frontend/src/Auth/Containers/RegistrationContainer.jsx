import React from "react";

import RegistrationForm from "Auth/Components/RegistrationForm";
import {withRouter} from "react-router-dom";
import api from "utils/api";
import FormContainer from "Forms/Containers/FormContainer";

export class RegistrationContainer extends React.Component {

    constructor(props) {
        super(props);

        this.onSuccess = this.onSuccess.bind(this);
    }

    onSuccess(data) {
        this.props.history.push('/login')
    }

    render() {
        return (
            <FormContainer onSuccess={this.onSuccess} apiFunction={api.registerUser}>
                <RegistrationForm />
            </FormContainer>
        );
    }
}

export default withRouter(RegistrationContainer)