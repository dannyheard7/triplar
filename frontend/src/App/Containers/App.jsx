import React from "react";
import {Route, Router, Switch} from "react-router-dom";
import {DragDropContext} from 'react-dnd';
import HTML5Backend from 'react-dnd-html5-backend';
import {connect} from "react-redux";
import ReactGA from 'react-ga';
import {Helmet} from "react-helmet";

import {history} from "../../store";
import {trackingId} from "../utils/constants";

import LoggedInContainer from "App/Containers/LoggedInContainer";
import HeaderAuthenticated from "App/Components/HeaderAuthenticated";
import HeaderUnauthenticated from "App/Components/HeaderUnauthenticated";
import Footer from "App/Components/Footer";
import LoginContainer from "Auth/Containers/LoginContainer";
import RegistrationContainer from "Auth/Containers/RegistrationContainer";
import TripCreateContainer from "Trips/Components/TripCreate";
import TripEditContainer from "../../Trips/Components/TripEdit";
import TripDelete from "../../Trips/Components/TripDelete";
import PlaceDetail from "../../Places/Components/PlaceDetail";
import {getCookie} from "../utils/utils";
import {tokenRefresh} from "../../Auth/utils/actions";
import TripDetail from "../../Trips/Components/TripDetail";
import TripList from "../../Trips/Components/TripList";
import ItinerariesOverview from "../../Itinerary/Components/ItinerariesOverview";
import LocationItinerary from "../../Itinerary/Components/LocationItinerary";

import 'App/styles/app.css';

export class App extends React.Component {
    componentDidMount() {
        if(!this.props.auth.successful) {
            const loginToken = getCookie('userToken');

            if (loginToken) {
                this.props.dispatch(tokenRefresh(loginToken));
            }
        }

        this.initializeReactGA();
    }

    initializeReactGA() {
        ReactGA.initialize('UA-130249377-1');
        ReactGA.pageview("/");
        history.listen(location => ReactGA.pageview(location.pathname));
    }

    render() {
        if(this.props.auth.successful) {
            return (
                <Router history={history}>
                    <LoggedInContainer token={this.props.auth.token}>
                        <Helmet>
                            <title>Triplar</title>
                        </Helmet>
                        <div className="wrapper">
                            <HeaderAuthenticated/>
                            <div className="content">
                                <Route path="/(|trips)" exact component={TripList}/>
                                <Route path="/(|trips)" exact component={TripCreateContainer}/>

                                <Route path="/trips/:id" component={TripDetail}/>
                                <Route path="/trips/:id/edit" component={TripEditContainer}/>
                                <Route path="/trips/:id/delete" component={TripDelete} />

                                <Route path="/trips/:id" component={ItinerariesOverview}/>

                                <Route path="/trips/:tripId/itinerary/:itineraryId" component={LocationItinerary}/>

                                <Route path="*/place/:placeId" component={PlaceDetail}/>
                            </div>
                            <Footer/>
                        </div>
                    </LoggedInContainer>
                </Router>
            );
        } else {
            return (
                <Router history={history}>
                    <div className="wrapper">
                        <Helmet>
                            <title>Triplar</title>
                        </Helmet>
                        <HeaderUnauthenticated/>
                        <div className="content">
                            <Switch>
                                <Route path="/register" exact component={RegistrationContainer}/>
                                <Route path="/" component={LoginContainer}/>
                            </Switch>
                        </div>
                        <Footer/>
                    </div>
                </Router>
            );
        }
    }
}

const mapStateToProps = state => {
    if(state.auth.successful) {
        return {
            auth: {...state.auth, token: getCookie('userToken')}
        }
    } else {
        return {auth: state.auth}
    }
};

const connectedApp = connect(mapStateToProps)(App);

export default DragDropContext(HTML5Backend)(connectedApp)