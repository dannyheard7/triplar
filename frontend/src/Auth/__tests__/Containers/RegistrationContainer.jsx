import React from "react";
import {shallow} from "enzyme";
import ShallowRenderer from 'react-test-renderer/shallow';

import { RegistrationContainer } from "Auth/Containers/RegistrationContainer";
import FormContainer from "Forms/Containers/FormContainer";

const renderer = new ShallowRenderer();
const faker = require('faker');

describe('<RegistrationContainer />', () => {
    const data = {user: {email: faker.internet.email(), password: faker.internet.password(), first_name: faker.name.firstName(),
        last_name: faker.name.lastName()}};
    const props = {
        history: {push: jest.fn()}
    };

    afterEach(() => {
       props.history.push.mockReset();
    });

    test('renders correctly', () => {
        const result = renderer.render(<RegistrationContainer {...props} />);
        expect(result).toMatchSnapshot();
    });

    test('calls onSuccess after receiving success from <FormContainer />',  () => {
        const spy = jest.spyOn(RegistrationContainer.prototype, "onSuccess");
        const container = shallow(<RegistrationContainer  {...props}/>);

        container.find(FormContainer).prop('onSuccess')(data);

        expect(spy).toBeCalled();
    });

    test('redirection to login after successful registration', () => {
        const container = shallow(<RegistrationContainer {...props} />);
        container.instance().onSuccess(data);

        expect(props.history.push).toBeCalledWith('/login');
    });
});