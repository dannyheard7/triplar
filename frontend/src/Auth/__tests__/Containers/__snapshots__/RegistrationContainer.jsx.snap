// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegistrationContainer /> renders correctly 1`] = `
<FormContainer
  apiFunction={[Function]}
  onSuccess={[Function]}
>
  <RegistrationForm />
</FormContainer>
`;
